MoreLinq.Test/OrderedMergeTest.cs

```diff
@@ -0,0 +1,294 @@
+
+            Assert.That(merged, Is.EqualTo(new[] { 2, 3, 1 }));
+        }
+
+        [TestFixture]
+        public class OverloadCorrectness
+        {
+            [Test]
+            public void first_second()
+            {
+                var first = new[] { 1, 4 };
+                var second = new[] { 2, 3 };
+
+                var overload = first.OrderedMerge(second);
+
+                var fullOrderedMerge = first.OrderedMerge(second,
+                    firstKeySelector: value => value,
+                    secondKeySelector: value => value,
+                    firstSelector: value => value,
+                    secondSelector: value => value,
+                    bothSelector: (f, _) => f,
+                    comparer: null);
+
+                Assert.That(overload, Is.EqualTo(fullOrderedMerge));
+                Assert.That(fullOrderedMerge, Is.EqualTo(new [] { 1, 2, 3, 4 }));
+            }
+
+            [Test]
+            public void first_second_comparer()
+            {
+                var first = new[] { 1, 4 };
+                var second = new[] { 2, 3 };
+                var comparer = Comparer.Create<int>((x, y) => y.CompareTo(x));
+
+                var overload = first.OrderedMerge(second,
+                    comparer);
+
+                var fullOrderedMerge = first.OrderedMerge(second,
+                    firstKeySelector: value => value,
+                    secondKeySelector: value => value,
+                    firstSelector: value => value,
+                    secondSelector: value => value,
+                    bothSelector: (f, _) => f,
+                    comparer: comparer);
+
+                Assert.That(overload, Is.EqualTo(fullOrderedMerge));
+                Assert.That(fullOrderedMerge, Is.EqualTo(new[] { 2, 3, 1, 4 }));
+            }
+
+            [Test]
+            public void first_second_keySelector()
+            {
+                var first = new[] { 4, 1 }; // 2, 8,
+                var second = new[] { 8, 2 }; // 1, 4
+                int KeySelector(int key) => 8 / key;
+
+                var overload = first.OrderedMerge(second,
+                    KeySelector);
+
+                var fullOrderedMerge = first.OrderedMerge(second,
+                    firstKeySelector: KeySelector,
+                    secondKeySelector: KeySelector,
+                    firstSelector: value => value,
+                    secondSelector: value => value,
+                    bothSelector: (f, _) => f,
+                    comparer: null);
+
+                Assert.That(overload, Is.EqualTo(fullOrderedMerge));
+                Assert.That(fullOrderedMerge, Is.EqualTo(new[] { 8, 4, 2, 1 }));
+            }
+
+            [Test]
+            public void first_second_keySelector_firstSelector_secondSelector_bothSelector()
+            {
+                var first = new[] { 1, 4 };
+                var second = new[] { 2, 3 };
+                int KeySelector(int id) => id;
```

Comment:
Is there a reason you decided not to introduce [an identity function like suggested like suggested the last time](https://github.com/morelinq/MoreLINQ/pull/486#discussion_r188924218)?

```c#
static T Id<T>(T id) => id;
```

It would remove the need for all local functions like `KeySelector`, `FirstSelector`, `SecondSelector` here and elsewhere, as well as many instances of `value => value` or `id => id` in other test cases.

Reply:
Yes, for the Overload examples I left them in as a documentation point. The actual functions don't matter so much as showing the equivalent functionality in full parameter version.

We can inline them and see what it looks like but initially I think it'll look worse for it - willing to give it a go and see.

Reply:
I reckon it'll read just fine since the named argument serve as the “documentation point”:

```c#
            [Test]
            public void first_second_keySelector_firstSelector_secondSelector_bothSelector()
            {
                var first = new[] { 1, 4 };
                var second = new[] { 2, 3 };
                int SelectFirst(int f, int _) => f;

                var overload = first.OrderedMerge(second,
                    keySelector: Id,
                    firstSelector: Id,
                    secondSelector: Id,
                    bothSelector: SelectFirst);

                var fullOrderedMerge = first.OrderedMerge(second,
                    firstKeySelector: Id,
                    secondKeySelector: Id,
                    firstSelector: Id,
                    secondSelector: Id,
                    bothSelector: SelectFirst,
                    comparer: null);

                Assert.That(overload, Is.EqualTo(fullOrderedMerge));
                Assert.That(fullOrderedMerge, Is.EqualTo(new[] { 1, 2, 3, 4 }));
            }
```

I'd also suggest renaming `BothSelector` to `SelectFirst` (alternatively `SelectFirstFromBoth`) to reflect _what it does_ as opposed to _what it is_. The argument annotation, `bothSelector:`, tells us what it is already. Ditto for other test cases.